### PR TITLE
fix(liveness): Multiple Liveness Bug Fixes

### PR DESCRIPTION
**Pull request by tylerjroach**
- [ ] PR conforms to [Pull Request](https://github.com/aws-amplify/amplify-ui-android/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guidelines.

*Issue #, if available:*

*Description of changes:*

* [Prevent edge case of multiple success/failure callbacks](https://github.com/aws-amplify/amplify-ui-android/commit/91002125da2fcba4df0ed4f88b02377c8c1c07d7)
* [Attempted fix which could result in wrong liveness version being passed.](https://github.com/aws-amplify/amplify-ui-android/commit/9c2cf6dd550c6a31ba4214a8d43cc2c8c4ed9e98)
* [Prevent crash if frame fails to analyze](https://github.com/aws-amplify/amplify-ui-android/commit/f942c6a3f1887461f6830da2f22699a2f41fd57b)
* [Prevent crash if liveness muxer fails to initialize](https://github.com/aws-amplify/amplify-ui-android/commit/794619337697439a33cf52870a33dc099ca8d8d0)
* [prevent crash if muxer fails to write encoded chunk](https://github.com/aws-amplify/amplify-ui-android/commit/2fa1d4e3e243b8ffbb122e6b7342932342cde969)

*How did you test these changes?*
(Please add a line here how the changes were tested)

*Documentation update required?*
- [ ] No
- [ ] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [ ] Added Unit Tests
- [ ] Added Integration Tests
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ankpshah — DISMISSED** [review]

File: liveness/src/main/java/com/amplifyframework/ui/liveness/camera/FrameAnalyzer.kt [hunk 1]
```diff
@@ -42,7 +43,17 @@ internal class FrameAnalyzer(
     private var cachedBitmap: Bitmap? = null
     private var faceDetector = FaceDetector(livenessState)
 
+    private val logger = Amplify.Logging.forNamespace("Liveness")
+
     override fun analyze(image: ImageProxy) {
+        try {
+            attemptAnalyze(image)
+        } catch (e: Exception) {
+            logger.error("Failed to analyze frame", e)
```
Comment (mattcreaser) [inline liveness/src/main/java/com/amplifyframework/ui/liveness/camera/FrameAnalyzer.kt:52]: Let's add a comment here around our understanding of this exception and why we are intentionally dropping a frame

File: liveness/api/liveness.api [hunk 2]
```diff
@@ -2,6 +2,7 @@ public final class com/amplifyframework/ui/liveness/BuildConfig {
 	public static final field BUILD_TYPE Ljava/lang/String;
 	public static final field DEBUG Z
 	public static final field LIBRARY_PACKAGE_NAME Ljava/lang/String;
+	public static final field LIVENESS_VERSION_NAME Ljava/lang/String;
```
Comment (mattcreaser) [inline liveness/api/liveness.api:5]: To consider: should we be omitting BuildConfig from our API validation?

File: liveness/src/main/java/com/amplifyframework/ui/liveness/camera/LivenessMuxer.kt [hunk 3]
```diff
@@ -61,7 +64,11 @@ internal class LivenessMuxer(
             }
         }
 
-        muxer.writeSampleData(videoTrack, byteBuf, bufferInfo)
+        try {
+            muxer.writeSampleData(videoTrack, byteBuf, bufferInfo)
+        } catch (e: Exception) {
+            logger.error("Failed to write encoded chunk to muxer", e)
```
Comment (mattcreaser) [inline liveness/src/main/java/com/amplifyframework/ui/liveness/camera/LivenessMuxer.kt:70]: Add comment here

File: liveness/src/main/java/com/amplifyframework/ui/liveness/camera/LivenessVideoEncoder.kt [hunk 4]
```diff
@@ -152,11 +154,16 @@ internal class LivenessVideoEncoder private constructor(
...
-                            )
+                            try {
+                                val muxer = LivenessMuxer(
+                                    outputFile,
+                                    encoder.outputFormat,
+                                    onMuxedSegment
+                                )
+                                livenessMuxer = muxer
+                            } catch (e: Exception) {
+                                logger.error("Failed to create liveness muxer", e)
```
Comment (mattcreaser) [inline liveness/src/main/java/com/amplifyframework/ui/liveness/camera/LivenessVideoEncoder.kt:165]: Comment here

File: liveness/api/liveness.api (hunk 2 above)
Comment (tylerjroach) [reply liveness/api/liveness.api:5]: Worth a discussion, but I don't think this decision needs to be made just yet.

**Review by ankpshah — APPROVED** [review]

**Pull request merged**